### PR TITLE
reject historical place partitioning and print to screen if all are r…

### PR DESCRIPTION
**Pull request by kkdavis14**
(if all are rejected)

it's ugly but it should work. any tips on how I can tighten this up? or if there's a better approach entirely. this is what I've come up with. 

rejecting any prop with P582 is easy, it's the printing to screen if they're all rejected that adds all the code.

closes https://git.yale.edu/LUX/pipeline/issues/668

### CODE REVIEW

File: pipeline/sources/general/wikidata/fetcher.py [hunk 1]
```diff
@@ -34,6 +34,8 @@ def post_process(self, js, identifier):
         new['altLabel'] = altLbls
         new['description'] = descs
 
+        part_ofs = {}
```
Comment (azaroth42) [inline pipeline/sources/general/wikidata/fetcher.py:37]: report = False

File: pipeline/sources/general/wikidata/fetcher.py [hunk 2]
```diff
@@ -44,6 +46,21 @@ def post_process(self, js, identifier):
                 if 'qualifiers' in val and 'P2241' in val['qualifiers']:
                     # deprecated reason, but not deprecated rank
                     continue
+                if prop in ['P131','P17']:
+                    if 'qualifiers' in val and 'P582' in val['qualifiers']:
+                        if js['id'] in part_ofs:
+                            part_ofs[js['id']].append({prop:False})
+                        else:
+                            part_ofs[js['id']] = [{prop:False}]
+                        #record it has a prop and a bad qualifier
```
Comment (azaroth42) [inline pipeline/sources/general/wikidata/fetcher.py:55]: report = True

File: pipeline/sources/general/wikidata/fetcher.py [hunk 3]
```diff
@@ -44,6 +46,21 @@ def post_process(self, js, identifier):
...
+                        #don't add it to newvals
+                        continue
+                    else:
+                        #record it has a prop and it can go through
```
Comment (azaroth42) [inline pipeline/sources/general/wikidata/fetcher.py:59]: report = False

File: pipeline/sources/general/wikidata/fetcher.py [hunk 4]
```diff
@@ -70,4 +87,11 @@ def post_process(self, js, identifier):
                     continue
                 newvals.append(dvv)
             new[prop] = newvals
+
+        for k, v in part_ofs.items():
```
Comment (azaroth42) [inline pipeline/sources/general/wikidata/fetcher.py:91]: if report:
  print(f"{js['id']} has only parts with ends via P582")

**Review by azaroth42 — CHANGES_REQUESTED** [review]
Agree it's a bit messy. Especially for just debugging whether we need to care or not.

Proposed an easier solution

**Comment by kkdavis14** [comment]
glad it took me 3 hours to figure out the overly complicated solution (/lays head on desk)

**Comment by azaroth42** [comment]
It's learning which tools of the language to use in which situations, and that comes only with time and practice :)

**Comment by azaroth42** [comment]
Almost -- needs to deal with the case where there are one or more entries without a qualifier, but the last one has one (which will set to True but shouldn't)

**Comment by kkdavis14** [comment]
not convinced this works.

File: pipeline/sources/general/wikidata/fetcher.py [hunk 5]
```diff
@@ -70,4 +78,8 @@ def post_process(self, js, identifier):
                     continue
                 newvals.append(dvv)
             new[prop] = newvals
+
+        if all(str(report)) == True:
```
Comment (azaroth42) [inline pipeline/sources/general/wikidata/fetcher.py:82]: no need for str() here :)

**Review by azaroth42 — APPROVED** [review]

**Pull request merged**